### PR TITLE
hard to know the users personal Vim Runtime Directory

### PR DESCRIPTION
**Pull request by chrisbra**
Problem:  hard to guess the Vim Runtime Directory
Solution: Set the $MYVIMRCDIR environment variable to the directory
          containing $MYVIMRC

### CODE REVIEW

File: runtime/doc/starting.txt [hunk 1]
```diff
@@ -828,8 +828,8 @@ accordingly.  Vim proceeds in this order:
 	easy to copy it to another system.
 
 	If Vim was started with "-u filename", the file "filename" is used.
-	All following initializations until 4. are skipped. $MYVIMRC is not
-	set.
+	All following initializations until 4. are skipped. $MYVIMRC and
+	$MYVIMRCDIR is not set.
```
Comment (habamax) [inline runtime/doc/starting.txt]: are not set?

Comment (chrisbra) [reply runtime/doc/starting.txt]: yeah correct

**Comment by k-takata** [comment]
Just curious, but what is the use case that $MYVIMRC is not enough?

**Comment by habamax** [comment]
My guess is to (or at least I would have used it) to simplify following:

```
&directory = $'{fnamemodify($MYVIMRC, ":p:h")}/.data/swap/'
&backupdir = $'{fnamemodify($MYVIMRC, ":p:h")}/.data/backup//'
&undodir = $'{fnamemodify($MYVIMRC, ":p:h")}/.data/undo//'
if !isdirectory(&undodir)   | mkdir(&undodir, "p")   | endif
if !isdirectory(&backupdir) | mkdir(&backupdir, "p") | endif
if !isdirectory(&directory) | mkdir(&directory, "p") | endif
```

**Comment by chrisbra** [comment]
> &directory = $'{fnamemodify($MYVIMRC, ":p:h")}/.data/swap/'

basically that, except for that this doesn't work properly, if you are using `~/.vimrc` file, in which case `$MYVIMRCDIR` should point to `~/.vim/`.

I need to update the tests for this but at the same time busy with a few more security related issues :/

**Comment by jmdevin** [comment]
> basically that, except for that this doesn't work properly, if you are using ~/.vimrc file, in which case $MYVIMRCDIR should point to ~/.vim/.

If I understand that correctly, perhaps the new variable is better named `$MYVIMDIR` rather than `$MYVIMRCDIR`? Just from the name, I would expect `$MYVIMRCDIR` to resolve to `~` when the user's vimrc is at `~/.vimrc`.

**Comment by chrisbra** [comment]
> If I understand that correctly, perhaps the new variable is better named $MYVIMDIR rather than $MYVIMRCDIR

:) I had it as `$MYVIMDIR` initially, but I thought `MYVIMRCDIR` is clearer. I'll change it back then.

**Pull request closed without merging**